src/modal-manager.ts

```diff
@@ -0,0 +1,226 @@
+  /**
+   * The current mode of the ModalManager
+   *
+   * Current options are `modal` or `closed`
+   *
+   * @type {ModalManagerMode}
+   * @memberof ModalManager
+   */
+  @property({ type: String, reflect: true }) mode: ModalManagerMode =
+    ModalManagerMode.Closed;
+
+  /**
+   * Custom content to display in the modal's content slot
+   *
+   * @type {(TemplateResult | undefined)}
+   * @memberof ModalManager
+   */
+  @property({ type: Object }) customModalContent?: TemplateResult;
+
+  /**
+   * Reference to the ModalTemplate DOM element
+   *
+   * @private
+   * @type {ModalTemplate}
+   * @memberof ModalManager
+   */
+  @query('modal-template') private modalTemplate!: ModalTemplate;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <div class="container">
+        <div class="backdrop" @click=${this.backdropClicked}></div>
+        <modal-template @closeButtonPressed=${this.closeButtonPressed}>
+          ${this.customModalContent}
+        </modal-template>
+      </div>
+    `;
+  }
+
+  /** @inheritdoc */
+  getMode(): ModalManagerMode {
+    return this.mode;
+  }
+
+  /** @inheritdoc */
+  closeModal(): void {
+    this.mode = ModalManagerMode.Closed;
+  }
+
+  /**
+   * Whether the modal should close if the user taps on the backdrop
+   *
+   * @private
+   * @memberof ModalManager
+   */
+  private closeOnBackdropClick = true;
+
+  /**
+   * A callback if the user closes the modal
+   *
+   * @private
+   * @memberof ModalManager
+   */
+  private userClosedModalCallback?: () => void;
+
+  /**
+   * Call the userClosedModalCallback and reset it if it exists
+   *
+   * @private
+   * @memberof ModalManager
+   */
+  private callUserClosedModalCallback(): void {
+    // we assign the callback to a temp var and undefine it before calling it
+    // otherwise, we run into the potential for an infinite loop if the
+    // callback triggers another `showModal()`, which would execute `userClosedModalCallback`
+    const callback = this.userClosedModalCallback;
+    this.userClosedModalCallback = undefined;
+    if (callback) callback();
+  }
+
+  /** @inheritdoc */
+  async showModal(options: {
+    config: ModalConfig;
+    customModalContent?: TemplateResult;
+    userClosedModalCallback?: () => void;
+  }): Promise<void> {
+    this.closeOnBackdropClick = options.config.closeOnBackdropClick;
+    this.userClosedModalCallback = options.userClosedModalCallback;
+    this.modalTemplate.config = options.config;
+    this.customModalContent = options.customModalContent;
+    this.mode = ModalManagerMode.Modal;
+  }
+
+  /** @inheritdoc */
+  updated(changed: PropertyValues): void {
+    /* istanbul ignore else */
+    if (changed.has('mode')) {
+      this.emitModeChangeEvent();
+    }
+  }
+
+  /**
+   * Called when the backdrop is clicked
+   *
+   * @private
+   * @memberof ModalManager
+   */
+  private backdropClicked(): void {
+    if (this.closeOnBackdropClick) {
```

Comment:
You didn't need an `/* istanbul ignore else */` here as well?

Reply:
Doesn't seem to need it